node/import_map.ts

```diff
@@ -64,79 +104,53 @@ class ImportMap {
+        const path = fileURLToPath(url)
+        const relativePath = relative(this.rootPath, path)
 
-        return
+        if (relativePath.startsWith('..')) {
+          throw new Error(
+            `Import map cannot reference '${path}' as it's outside of the base directory '${this.rootPath}'`,
+          )
+        }
       }
 
-      resolvedImports[specifier] = url.toString()
+      filteredImports[specifier] = url.toString()
     })
 
-    return resolvedImports
+    return filteredImports
   }
 
-  // Takes a URL, turns it into a path relative to the given base, and prepends
-  // a prefix (such as the virtual root prefix).
-  static resolvePath(url: URL, basePath?: string, prefix?: string) {
-    if (basePath === undefined) {
-      return url.toString()
-    }
+  // Takes a `scopes` object and runs all imports through `filterImports`,
+  // omitting any scopes for which there are no imports.
+  filterScopes(scopes?: ParsedImportMap['scopes']) {
+    const filteredScopes: Record<string, Imports> = {}
 
-    const path = fileURLToPath(url)
-    const relativePath = relative(basePath, path)
-
-    if (relativePath.startsWith('..')) {
-      throw new Error(`Import map cannot reference '${path}' as it's outside of the base directory '${basePath}'`)
-    }
+    if (scopes !== undefined) {
+      Object.keys(scopes).forEach((url) => {
+        const imports = this.filterImports(scopes[url])
 
-    // We want to use POSIX paths for the import map regardless of the OS
-    // we're building in.
-    let normalizedPath = relativePath.split(sep).join(posix.sep)
-
-    // If the original URL had a trailing slash, ensure the normalized path
-    // has one too.
-    if (normalizedPath !== '' && url.pathname.endsWith(posix.sep) && !normalizedPath.endsWith(posix.sep)) {
-      normalizedPath += posix.sep
-    }
-
-    const newURL = new URL(normalizedPath, prefix)
-
-    return newURL.toString()
-  }
-
-  add(source: ImportMapSource) {
-    this.sources.push(source)
-  }
-
-  async addFile(path: string, logger: Logger) {
-    const source = await readFile(path, logger)
+        if (Object.keys(imports).length === 0) {
+          return
+        }
 
-    if (Object.keys(source.imports).length === 0) {
-      return
+        filteredScopes[url] = imports
+      })
     }
 
-    return this.add(source)
+    return filteredScopes
   }
 
-  async addFiles(paths: (string | undefined)[], logger: Logger) {
-    for (const path of paths) {
-      if (path === undefined) {
-        continue
-      }
-
-      await this.addFile(path, logger)
-    }
-  }
-
-  getContents(basePath?: string, prefix?: string) {
+  // Returns the import map as a plain object, with any relative paths resolved
+  // to full URLs. It takes an optional `prefixes` object that specifies a list
+  // of prefixes to replace path prefixes (see `applyPrefixesToPath`). Prefixes
+  // will be applied on both `imports` and `scopes`.
+  getContents(prefixes: Record<string, string> = {}) {
```

Comment:
Previously, `getContents` accepted a base path parameter, which doesn't really make sense because all paths are already absolute before this method returns.

What we want is the ability to transform the paths to use different prefixes (such as the `file:///root/` virtual root for ESZIPs), so this method now accepts an optional `prefixes` object. So far we have one prefix, but we'll likely have more in the future.